cxx-qt-gen/src/extract.rs

```diff
@@ -330,6 +338,11 @@ fn extract_type_ident(
     // Extract the Qt type this is used in C++ and Rust generation
     let qt_type = extract_qt_type(&idents, ty, cpp_namespace_prefix, qt_ident)?;
 
+    // Check if this Qt type is allowed to be a ref mut
+    if is_mut && is_ref && !qt_type.ref_mut_is_valid() {
+        return Err(ExtractTypeIdentError::InvalidType(ty.span()));
```

Comment:
InvalidType here is a bit of a stretch. But I think we'll need to spend some time on better compiler errors anyway. So this is something for later.

Reply:
Right, i think we'll have a round of improving error messages later.